tree/treeplayer/src/TDFUtils.cxx

```diff
@@ -152,6 +153,45 @@ unsigned int GetNSlots()
    return nSlots;
 }
 
+void GetBranchNamesImpl(TTree *t, std::set<std::string> &bNames, std::set<TTree *> &analysedTrees)
+{
+
+   if (!t)
+      return;
+
+   if (!analysedTrees.insert(t).second) {
+      return;
+   }
+
+   auto branches = t->GetListOfBranches();
+   if (branches) {
+      for (auto branchObj : *branches) {
+         bNames.insert(branchObj->GetName());
+      }
+   }
+
+   auto friendTrees = t->GetListOfFriends();
+
+   if (!friendTrees)
+      return;
+
+   for (auto friendTreeObj : *friendTrees) {
+      auto friendTree = ((TFriendElement *)friendTreeObj)->GetTree();
+      GetBranchNamesImpl(friendTree, bNames, analysedTrees);
+   }
+}
+
+ColumnNames_t GetBranchNames(TTree *t)
```

Comment:
we shouldn't call this function if there is no tree, so the `TTree*` could be changed to a `TTree&`

Reply:
Despite this being an internal helper the comment makes sense.